### ISSUE TITLE
[commonjs--resolver] data URI with non-JavaScript mime type is not supported.

### ISSUE THREAD

**Issue by emosheeep** [laynezh/vite-plugin-lib-assets#34]
<img width="1817" alt="image" src="https://github.com/laynezh/vite-plugin-lib-assets/assets/31957758/301b9277-1f1a-47db-8a4a-2263910d41d6">

Don't know why...

**Comment by coder-layne** [comment]
Thanks for your feedback. I can't get enough information from the error message to identify the source of the exception. Can you provide a minimal reproducible version?

**Comment by emosheeep** [comment]
emmm...maybe i caught the error
<img width="1332" alt="image" src="https://github.com/laynezh/vite-plugin-lib-assets/assets/31957758/769be012-1ca9-4e6b-8b93-477db235d893">
This plugin didn't support base64 string to be embedded in style files.

**Comment by emosheeep** [comment]
And when I remove base64 and continue building, this error occurs
<img width="1541" alt="image" src="https://github.com/laynezh/vite-plugin-lib-assets/assets/31957758/b23c362d-9a0a-46e3-8d47-a5a5848cec8d">
I still don't know why 😭

**Comment by coder-layne** [comment]
> emmm...maybe i caught the error <img alt="image" width="1332" src="https://user-images.githubusercontent.com/31957758/282232118-769be012-1ca9-4e6b-8b93-477db235d893.png"> This plugin didn't support base64 string to be embedded in style files.

@emosheeep I'll look into this issue soon.

**Comment by coder-layne** [comment]
> emmm...maybe i caught the error <img alt="image" width="1332" src="https://user-images.githubusercontent.com/31957758/282232118-769be012-1ca9-4e6b-8b93-477db235d893.png"> This plugin didn't support base64 string to be embedded in style files.

@emosheeep In the latest v0.5.7, the issue you encountered has been resolved.

**Comment by emosheeep** [comment]
Sounds wonderful. I'll try tomorrow.

**Comment by emosheeep** [comment]
> And when I remove base64 and continue building, this error occurs
> <img alt="image" width="1541" src="https://user-images.githubusercontent.com/31957758/282232402-b23c362d-9a0a-46e3-8d47-a5a5848cec8d.png">
> **This is my config**
> <img alt="image" width="609" src="https://user-images.githubusercontent.com/31957758/282232500-7a1c891e-e425-45e6-9115-a2c9ece183af.png">
> I still don't know why 😭

Now the problem became this....

**Comment by coder-layne** [comment]
@emosheeep I'll try to reproduce the issue, identify its cause, and fix it if possible.

**Comment by emosheeep** [comment]
Thanks a lot！🫶

**Comment by coder-layne** [comment]
@emosheeep Sorry for the delayed response. I was failed to reproduce this issue. Can you provide a minimal reproducible version?

**Comment by showonne** [comment]
My project import katex css

```css
import 'katex/dist/katex.min.css';
```

and include font assets:

```js
export const KNOWN_ASSET_TYPES = [
  // fonts
  'woff2?',
  'eot',
  'ttf',
  'otf',
]

export const DEFAULT_ASSETS_RE = new RegExp(
  `\\.(` + KNOWN_ASSET_TYPES.join('|') + `)(\\?.*)?$`,
);

// config options
libAssets({
  name: '[name].[contenthash:8].[ext]?[query]',
  include: DEFAULT_ASSETS_RE
})
```

have same error:
<img width="464" alt="image" src="https://github.com/laynezh/vite-plugin-lib-assets/assets/9353213/13fcd63f-9792-4721-9730-1e2bcd79e920">

**Comment by showonne** [comment]
seems some error occured in `getAssetContent`

```js
content = fs.readFileSync(pureId);
```

when the `pureId` is a relative path，content will be null

**Comment by coder-layne** [comment]
@showonne Thanks for your feedback. With your guidance, I've successfully replicated the issue. Now, I'll try to find the cause and resolving it.

**Comment by coder-layne** [comment]
@emosheeep After testing, I found that `vite-plugin-svelte` also has a similar issue. The reason is that when using `preprocessCSS` (internally using `postcss`) to concatenate files, the URL addresses in the imported files are not updated to the relative path of the entry file.

There is also an [issue](https://github.com/vitejs/vite/issues/7651) in the Vite repository similar to the mentioned problem.

I plan to use sourcemaps to complete this update and will soon create a demo to confirm its effectiveness.

**Comment by coder-layne** [comment]
@emosheeep I just released version v0.5.13-beta.0, hoping it can solve the issues you encountered.

**Comment by emosheeep** [comment]
<img width="1518" alt="image" src="https://github.com/laynezh/vite-plugin-lib-assets/assets/31957758/3883cc22-16aa-437f-9a45-8527c813b9fc">

I got this error

<img width="1110" alt="image" src="https://github.com/laynezh/vite-plugin-lib-assets/assets/31957758/316038b3-3604-4b96-974d-3aac52e6cdd1">

**Comment by coder-layne** [comment]
@emosheeep It seems that we have overcome the most difficult part, and now this issue appears to be a omission from the previous [PR](https://github.com/laynezh/vite-plugin-lib-assets/pull/52) fix.

**Comment by coder-layne** [comment]
@emosheeep I just released version v0.5.13-beta.1, I think it will solve the problem shown in picture you encountered.

**Comment by emosheeep** [comment]
Cool, the plugin works finally!
![Uploading 截屏2023-12-17 23.00.39.png…]()
